docs/04-api-reference/front-commerce-core/graphql.mdx

```diff
@@ -0,0 +1,435 @@
+```
+
+## `makeErrorLoggerInterceptor`
+
+Intercept and log errors
+
+```js
+makeErrorLoggerInterceptor({
+  isMaintenance: maintenance,
+  ignoreForbiddenErrors: ignore,
+});
+```
+
+Arguments:
+
+- `maintenance` (`function`): a function that recieve error as parameter and
+  return a `boolean` to tell if website is in maintenance mode
+- `ignore` (`boolean`): a boolean to tell if forbidden errors should be ignored
+
+_Example_
+
+```ts
+import { makeErrorLoggerInterceptor } from "@front-commerce/core/graphql";
+
+const errorLoggerInterceptor = makeErrorLoggerInterceptor({
+  isMaintenance: (error) =>
+    error.response &&
+    (error.response.status === 503 ||
+      (error.response.status === 500 &&
+        /maintenance mode .* enabled/.test(error.response.data))),
+  ignoreForbiddenErrors: ignoreForbiddenErrors,
+});
+```
+
+## `limitRateByClientIp`
+
+Limit a GraphQL mutation by client IP
+
+```ts
+limitRateByClientIp(config, resolver);
+```
+
+Arguments:
+
+- `config` (`object`): rate limiter configuration object:
+  - `max` (`number`): number of calls allowed within window
+  - `window` (`number`): window of maximum calls
+- `resolver` (`function`): GraphQL resolver
+
+_Example_
+
+```ts title="extension/acme/graphql/modules/contact/resolver.ts"
+import { limitRateByClientIp } from "@front-commerce/core/graphql";
+
+export default {
+  Mutation: {
+    sendContactEmail: limitRateByClientIp(
+      { max: 5, window: "1m" },
+      (_, { template, data, storeEmail }, { loaders }) =>
+        loaders.Contact.sendContactEmail(template, data, storeEmail)
+    ),
+  },
+};
+```
+
+## `reorderForIds`
+
+Reorder an array by ids
+
+```ts
+reorderForIds(ids, (key = "id"));
+```
+
+Arguments:
+
+- `ids` (`string[]` default to `id`): ids to reorder
+- `key` (`string`): key to use to order
+
+_Example_
+
+```ts
+import { reorderForIds } from "@front-commerce/core/graphql";
+
+const orderer = reorderForIds([2, 3, 1], "id");
+const reordered = orderer([{ id: 1 }, { id: 2 }, { id: 3 }]);
+```
+
+## `reorderForIdsCaseInsensitive`
+
+Same as [`reorderForIds`](#) but case insensitive
+
+## `makeBatchLoaderFromSingleFetch`
+
+Convert a single fetch to a batch loader
+
+```ts
+makeBatchLoaderFromSingleFetch(fetch, response);
+```
+
+Argumvents:
+
+- `fetch` (`function`): fetch function
+- `response` (`object`): response transformer
+
+_Example_
+
+```ts
+import { makeBatchLoaderFromSingleFetch } from "@front-commerce/core/graphql";
+
+const axiosInstance = axios.create({
+  baseURL: "https://acme.com/api",
+});
+
+const acmeLoader = makeBatchLoaderFromSingleFetch(
+  (depth) => axiosInstance.get(`/prodcuts`),
+  (response) => response.data.products
+);
+```
+
+## `maybeValue`
+
+Check if a value is defined, else print a debug message and return `null`
+
+```ts
+maybeValue(value, message);
+```
+
+Arguments:
+
+- `value` (`any`): value to check
+- `message` (`string`): debug message to print if value is not defined
```

Comment:
**issue:** the variable [is called `noValueDebugMessage` in the code](https://gitlab.blackswift.cloud/front-commerce/front-commerce/-/blob/main/packages/core/graphql/dataloaderHelpers.ts?ref_type=heads#L6). For better discoverability, it should also be the case in the documentation. It also helps greatly with the readability.

```suggestion
- `noValueDebugMessage` (`string`): debug message to print if value is not defined
```